src/main/java/com/akathist/maven/plugins/launch4j/Launch4jMojo.java

```diff
@@ -489,6 +492,33 @@ private void doExecute() throws MojoExecutionException {
             }
         }
     }
+    
+    private void fillSensibleHeaderTypeDefaults() throws MojoExecutionException {
+        if (headerType == null) {
+            if (classPath != null && classPath.mainClass != null) {
+                if (classPath.mainClass.toLowerCase().contains("gui")) {
+                    getLog().warn("headerType not set, defaulting to \"gui\" because the main class is named " + classPath.mainClass);
+                    headerType = "gui";
+                    return;
+                }
```

Comment:
I don't like such _magic_, it's can be confusing to users why it behaves like this.